File: Array_Hashing/1_Two_Sum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashTable = dict()

        for i in range(len(nums)):
            if target - nums[i] in hashTable:
                return [hashTable[target - nums[i]], i]
            else:
                hashTable[nums[i]] = i

File: Two_Pointers/167_Two_Sum_II/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0;
        int j = numbers.size() - 1;
        vector<int> ret;

        while(i<j) {
            int sum = numbers[i] + numbers[j];
            if(sum == target){
                ret.push_back(i + 1);
                ret.push_back(j + 1);
                break;
            }
            else if(sum < target){
                i++;
            }
            else {
                j--;
            }

        }
        return ret;
    }
};